cupyx/jit/_builtin_funcs.py

```diff
@@ -150,6 +150,54 @@ def call(self, env, array, index, value):
         return Data(f'{name}(&{target.code}, {value.code})', ctype)
 
 
+class Grid(BuiltinFunc):
+
+    def __call__(self, ndim):
+        """Compute the thread index in the grid.
+
+        Computation of the first integer is as follows::
+
+            jit.threadIdx.x + jit.blockIdx.x * jit.blockDim.x
+
+        and for the other two integers the ``y`` and ``z`` attributes are used.
+
+        Args:
+            ndim (int): The dimension of the grid. Only 1, 2, or 3 is allowed.
+
+        Returns:
+            int or tuple:
+                If ``ndim`` is 1, an integer is returned, otherwise a tuple.
+
+        .. note::
+            This function follows the convention of Numba's `numba.cuda.grid`_.
+
+        .. _numba.cuda.grid:
+            https://numba.readthedocs.io/en/stable/cuda/kernels.html#absolute-positions
+
+        """
+        super.__call__(self)
+
+    def call_const(self, env, ndim):
```

Comment:
@asi1024 Am I understanding it correctly that for `call_const` its arguments must be compile-time constants?

Reply:
Yes, that's right!